hy/core/language.hy

```diff
@@ -369,12 +369,15 @@
   "Chunks coll into n-tuples (pairs by default). The remainder, if any, is not
    included unless a fillvalue is specified. The step defaults to n, but can be
    more to skip elements, or less for a sliding window with overlap."
-  (setv
-   step (or step n)
-   slices (genexpr (itertools.islice coll start None step) [start (range n)]))
-  (if (is fillvalue -sentinel)
-    (apply zip slices)
-    (apply zip-longest slices {"fillvalue" fillvalue})))
+  (if (< n 1) [] (do
```

Comment:
Yes, let's raise an exception on negative n. `tee` already does that for us. (`islice` also ensures that `step` is positive.)

As for `n == 0`, I'm really not certain. If you actually _want_ an infinite sequence of empty tuples, you'd normally just `(repeat (,))`, so `partition` probably doesn't have to be able to do this. And diverging when you thought you had a finite iterator is not nice.

I'm inclined to just remove this check and let it return `[]` when `n == 0`, then see if anybody complains. @hylang/core thoughts? How should this work?